game/src/graphic/textures/TextureHandler.java

```diff
@@ -15,17 +18,44 @@
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>Singleton.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root_please_not_modify";
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        List<Path> placeholderPaths = findAllPlaceholderPaths();
+        assert placeholderPaths.size() == 2;
+        // takes the placeholder path with the longest path string...
+        if (placeholderPaths.get(0).toString().length()
+                > placeholderPaths.get(1).toString().length()) {
+            addAllAssets(new FileHandle(placeholderPaths.get(0).getParent().toFile()));
+        } else {
+            addAllAssets(new FileHandle(placeholderPaths.get(1).getParent().toFile()));
+        }
```

Comment:
das ist extrem unleserlich und fehleranfällig.

du verwendest die langen konstrukte je 2x, da könntest du dir wenigstens hilfsvariablen anlegen.

musst du wirklich noch toString() ausführen? hat ein path evtl. bereits eine length()-methode? (habe grad keine ide offen)

das finden des längsten pfads und das konvertieren in parent/file kannst du eigentlich elegant in dem stream in der neuen hilfsmethode mit erledigen und ein optional<file> zurück liefern.